zip-0202.rst

```diff
@@ -0,0 +1,262 @@
+
+Consider the following example of a serialized version 3 transaction.
+
+Pre-Overwinter parser:
+
+* data begins with little-endian byte sequence: [0x03, 0x00, 0x00, 0x80]
+* deserialized as 32-bit signed integer
+
+  * with hexadecimal value of 0x80000003 (most significant bit is set)
+  * decimal value of -2147483645
+
+Legacy parsers will expect the version to be a positive value, such as 1 or 2, and will thus reject the Overwinter transaction as invalid.
+
+Overwinter parser:
+
+- data begins with little-endian byte sequence: [0x03, 0x00, 0x00, 0x80]
+- deserialized as 32-bit unsigned integer
+
+  - with binary value of 0b10000000000000000000000000000011
+- the 32-bits are decomposed into two fields
+
+  - overwinter flag (bit 31) as a boolean, expected to be set
+  - version (bits 30 - bit 0) as an unsigned integer, expected to have a decimal value of 3
+
+Overwinter parsers will accept the transaction as valid as the most significant bit of the header has been set.  By masking off (unsetting) the most significant bit, the parser can retrieve the transaction version number::
+
+    0x80000003 & 0x7FFFFFFFF = 0x00000003 = 3
+
+Version Group Id
+----------------
+
+The version group id is a non-zero, random and unique identifier, of type uint32_t, assigned to a transaction format version, or a group of soft-forking transaction format versions.  The version group id helps nodes disambiguate between branches using the same version number.
+
+That is, it prevents a client on one branch of the network from attempting to parse transactions intended for another branch, in the situation where the transactions share the same format version number but are actually specified differently.  For example, Zcash and a clone of Zcash might both define their own custom v3 transaction formats, but each will have its own unique version group id, so that they can reject v3 transactions with unknown version group ids.
+
+The combination of transaction version and version group id, `nVersion || nVersionGroupId`, uniquely defines the transaction format, thus enabling parsers to reject transactions from outside the client's chain which cannot be parsed.
+
+By convention, it is expected that when introducing a new transaction version requiring a network upgrade, a new unique version group id will be assigned to that transaction version.
+
+However, if new transaction versions are soft-fork compatible with older transaction versions, the same version group id can be re-used.
+
+Expiry Height
+-------------
+
+The expiry height field, as defined in the Transaction Expiry ZIP [#zip-0203]_, stores the block height after which a transaction can no longer be mined.
+
+Transaction Validation
+======================
+
+A valid Overwinter transaction intended for Zcash MUST have:
+
+- version number 3
+- version group id 0x03C48270 [#versiongroupid]_
+- fOverwintered flag set
+
+Overwinter transaction parsers should reject transactions for violating consensus rules if:
```

Comment:
Technically the parser isn't responsible for rejecting transactions. It reports that it can't parse the transaction which, in the case where the parser is being used by a "full validator" [\*], causes the latter to reject the transaction &mdash; and this is a MUST, not a should/SHOULD. You could have a parser that isn't part of a full validator, and that wouldn't be required to reject the transaction (since "rejecting" transactions isn't something that such a parser does).

I suggest not talking about parsers here, and just saying "Overwinter validators MUST reject transactions for violating consensus rules if:"

[\*] The protocol spec, from 2018.0-beta-7 on, uses the term "full validator" for any entity that checks validity of the block chain against the full set of consensus rules. Full validators may or may not be nodes in the peer-to-peer network (they could get the block chain by some other means).